avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroDeserializer.java

```diff
@@ -95,10 +99,42 @@ public DatumReader<?> load(IdentityPair<Schema, Schema> key) {
    * Sets properties for this deserializer without overriding the schema registry client itself.
    * Useful for testing, where a mock client is injected.
    */
-  protected void configure(KafkaAvroDeserializerConfig config) {
+  protected void configure(KafkaAvroDeserializerConfig config, boolean isKey) {
```

Comment:
Let's mimic the Protobuf code instead.  There the `KafkaProtobufDeserializer` uses `isKey` and figures out what `Class` to pass to `AbstractKafkaProtobufDeserializer`.  So this `configure` method would take `Class<?> type` instead.